File: Week1_Design Patterns and Principles/Ex_2_Implementing_the_Factory_Method_Pattern/Code/com/example/factory/factories/PdfDocumentFactory.java
package com.example.factory.factories;
import com.example.factory.documents.Document;
import com.example.factory.documents.PdfDocument;
public class PdfDocumentFactory extends DocumentFactory {
    @Override
    public Document createDocument() {
        System.out.println("PdfDocumentFactory: Creating new PDF document...");
        return new PdfDocument();
    }
}

File: Week1_Design Patterns and Principles/Ex_2_Implementing_the_Factory_Method_Pattern/Code/Factories/PdfDocumentFactory.cs
using System;
using FactoryDemo.Documents;

namespace FactoryDemo.Factories;

public sealed class PdfDocumentFactory : DocumentFactory
{
    public override IDocument CreateDocument()
    {
        Console.WriteLine("PdfDocumentFactory: Creating new PDF document…");
        return new PdfDocument();
    }
}
